src/decays/experimental_constraints.cpp

```diff
@@ -0,0 +1,579 @@
+           status = gsl_min_fminimizer_iterate (sGSL);
+
+           m = gsl_min_fminimizer_x_minimum (sGSL);
+           a = gsl_min_fminimizer_x_lower (sGSL);
+           b = gsl_min_fminimizer_x_upper (sGSL);
+      }
+      while (std::abs(1. - sm.get_physical().Mhh/mass) > mass_precision && iter < max_iter);
+
+      gsl_min_fminimizer_free (sGSL);
+
+      if (const double diff = std::abs(1. - sm.get_physical().Mhh/mass); diff > mass_precision) {
+         throw std::runtime_error("Normalized Higgs effective couplings: Cannot find a SM equivalent of " + name +
+                     " after " + std::to_string(iter+1) + "/" + std::to_string(max_iter) + " iterations. "
+                     "Mass difference: " + std::to_string(mass) + " GeV (BSM) vs " +
+                     std::to_string(sm.get_physical().Mhh) + " GeV (SM) for λSM = " + std::to_string(m) +
+                     ". Difference: " + std::to_string(100*diff) + "%. ");
+      }
+
+
+}
+} // anonymous
+
+double chi2_to_pval(double chi2BSM, double chi2SM) {
+   boost::math::chi_squared dist(2);
+   const double pval = chi2BSM<=chi2SM ? 1 : boost::math::cdf(complement(dist, chi2BSM-chi2SM));
+   return pval;
+}
+
+EffectiveCoupling_list get_normalized_effective_couplings(
+   EffectiveCoupling_list const& bsm_input,
+   Physical_input const& physical_input,
+   softsusy::QedQcd const& qedqcd,
+   Spectrum_generator_settings const& spectrum_generator_settings,
+   FlexibleDecay_settings const& flexibledecay_settings)
+{
+   // make sure we don't compute input for the EFFHIGGSCOUPLINGS block in the
+   // built in SM
+   auto flexibledecay_settings_ = flexibledecay_settings;
+
+   EffectiveCoupling_list _bsm_input;
+   for (auto const& el : bsm_input) {
+
+      const double mass = el.mass;
+      // in the SM, λ = (mh/v)^2/2
+      // for mh > 700 GeV this gives λ > 4
+      // it probably makes no sense to use coupling strengh modifiers in this case so we skip those particles
+      // On the other hand there's a problem with finding a SM equivalent of very light states
+      if (mass > 650 || mass < 1) continue;
+
+      // create a SM equivalent to the BSM model, with mhSM == mass
+      standard_model::Standard_model sm {};
+      set_sm_settings_matching_bsm(sm, spectrum_generator_settings);
+      sm.set_physical_input(physical_input);
+      sm.initialise_from_input(qedqcd);
+      set_sm_lambda_to_match_bsm_mh(sm, mass, el.particle);
+      sm.calculate_pole_masses();
+
+      if (sm.get_physical().Mhh > 0) {
+         // calculate decays in the SM equivalent
+         flexiblesusy::Standard_model_decays sm_decays(sm, qedqcd, physical_input, flexibledecay_settings_);
+         sm_decays.calculate_decays();
+         const auto sm_input = sm_decays.get_neutral_higgs_effc();
+
+         // fermion channels are given as complex numbers
+         // we normalize to real part of SM coupling
+         // quarks
+         NeutralHiggsEffectiveCouplings _coups {el};
```

Comment:
Same comment here: Please do not use an underscore as first character for this local variable.

Reply:
Done in 167a4c3e1e4876f10692ce55b054c26b4fa921f3.